pkg/infra/infra.go

```
@@ -190,24 +209,135 @@ var _ = Describe("[sig-cluster-lifecycle] Machine API Managed cluster should", f
-			machineSetParams = framework.BuildMachineSetParams(ctx, client, 1)
+		BeforeAll(func() {
+			setupCtx := framework.GetContext()
+			setupClient, err := framework.LoadClient()
+			Expect(err).ToNot(HaveOccurred(), "Controller-runtime client should be able to be created for shared MachineSet setup")
 
-			By("Creating a new MachineSet")
+			params := framework.BuildMachineSetParams(setupCtx, setupClient, 1)
 
-			machineSet, err = framework.CreateMachineSet(client, machineSetParams)
+			By("Creating a shared 1-replica MachineSet")
+			sharedMachineSet, err = framework.CreateMachineSet(setupClient, params)
```

golangci-lint (2.11.4)

[error] missing whitespace above this line (invalid statement above assign)

(wsl_v5)